support/cas-server-support-rest-x509/src/main/java/org/apereo/cas/support/x509/rest/X509RestHttpRequestCredentialFactory.java

```diff
@@ -28,10 +31,30 @@
  * @author Dmytro Fedonin
  * @since 5.1.0
  */
+@RequiredArgsConstructor
 @Slf4j
 public class X509RestHttpRequestCredentialFactory implements RestHttpRequestCredentialFactory {
```

Comment:
It doesn't make much sense to combine the functionality implemented here with the old behavior in one class and then differentiate based on a setting. I would probably prefer to create separate classes, different forms of `RestHttpRequestCredentialFactory` that each do exactly one thing. Then in the bean definition, based on the presence of a setting or its value, we can create each object accordingly.

```java
if (oldWay) {
   return new X509RestHttpRequestCredentialFactory();
}

return new YourX509RestHttpRequestCredentialFactory(x509CertificateExtractor.getIfAvailable());
```

Of course, name the class such that its intentions are revealed. Don't be shy about getting wordy :) 

The drawback would be that you wouldn't be able to use both new and old strategies at the same time, but I can't imagine that would be a proper use case and IIF that somehow becomes one, we can always refactor at a later date.